pkg/cmd/admin/policy/review.go

```diff
@@ -0,0 +1,191 @@
+	$  %[1]s -f myresource_with_no_sa.yaml
+	`)
+)
+
+const ReviewRecommendedName = "review"
+
+type reviewOptions struct {
+	client              client.PodSecurityPolicyReviewsNamespacer
+	namespace           string
+	enforceNamespace    bool
+	out                 io.Writer
+	mapper              meta.RESTMapper
+	typer               runtime.ObjectTyper
+	RESTClientFactory   func(mapping *meta.RESTMapping) (resource.RESTClient, error)
+	printer             kubectl.ResourcePrinter
+	FilenameOptions     resource.FilenameOptions
+	ServiceAccountNames []string
+}
+
+func NewCmdReview(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	o := &reviewOptions{}
+	cmd := &cobra.Command{
+		Use:     name,
+		Short:   "Checks which ServiceAccount can create a Pod",
+		Long:    reviewLong,
+		Example: fmt.Sprintf(reviewExamples, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, args, cmd, out))
+			kcmdutil.CheckErr(o.Run(args))
+		},
+	}
+
+	cmd.Flags().StringSliceVarP(&o.ServiceAccountNames, "serviceaccounts", "s", o.ServiceAccountNames, "List of ServiceAccount names, comma separated")
+	kcmdutil.AddFilenameOptionFlags(cmd, &o.FilenameOptions, "Filename, directory, or URL to a file identifying the resource to get from a server.")
+
+	kcmdutil.AddPrinterFlags(cmd)
+
+	return cmd
+}
+
+func (o *reviewOptions) Complete(f *clientcmd.Factory, args []string, cmd *cobra.Command, out io.Writer) error {
+	if len(args) == 0 && len(o.FilenameOptions.Filenames) == 0 {
+		return kcmdutil.UsageError(cmd, cmd.Use)
+	}
+	var err error
+	o.namespace, o.enforceNamespace, err = f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	o.client, _, err = f.Clients()
+	if err != nil {
+		return fmt.Errorf("unable to obtain client: %v", err)
+	}
+	o.mapper, o.typer = f.Object()
+	o.RESTClientFactory = f.ClientForMapping
+
+	if len(kcmdutil.GetFlagString(cmd, "output")) != 0 {
+		clientConfig, err := f.ClientConfig()
+		if err != nil {
+			return err
+		}
+		version, err := kcmdutil.OutputVersion(cmd, clientConfig.GroupVersion)
+		if err != nil {
+			return err
+		}
+		p, _, err := kcmdutil.PrinterForCommand(cmd)
+		if err != nil {
+			return err
+		}
+		o.printer = kubectl.NewVersionedPrinter(p, kapi.Scheme, version)
+	} else {
+		o.printer = describe.NewHumanReadablePrinter(kubectl.PrintOptions{NoHeaders: kcmdutil.GetFlagBool(cmd, "no-headers")})
+	}
+
+	o.out = out
+
+	return nil
+}
+
+func (o *reviewOptions) Run(args []string) error {
+	allErrs := []error{}
+	reviews := []*securityapi.PodSecurityPolicyReview{}
+
+	infos, err := resource.NewBuilder(o.mapper, o.typer, resource.ClientMapperFunc(o.RESTClientFactory), kapi.Codecs.UniversalDecoder()).
+		NamespaceParam(o.namespace).
+		FilenameParam(o.enforceNamespace, &o.FilenameOptions).
+		ResourceTypeOrNameArgs(true, args...).
+		ContinueOnError().
+		Flatten().
+		Do().
+		Infos()
```

Comment:
Although I would not be opposed if not done this way, having this block return a `resource.Result`, and then using the result's `.Visitor(func...)` to iterate through each resource would match how this is [done in other commands](https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/kubectl/cmd/create.go#L118-L133).